.ci.sh

```diff
@@ -30,14 +30,7 @@ echo *** ANALYZE CODE ***
 pip install pylint
 bash -c 'PYTHONPATH="$PWD/scripts:$PWD" pylint -j3 --py3k --output-format=parseable --errors-only $(git ls-tree --name-only -r HEAD | sort | grep -E "(spinalcordtoolbox|scripts|testing).*\.py" | xargs); exit $(((($?&3))!=0))'
 
-#
-# echo *** BUILD DOCUMENTATION ***
-# pip install sphinx sphinxcontrib.programoutput sphinx_rtd_theme
-# cd documentation/sphinx
-# make html
-# cd -
-
-# python create_package.py -s ${TRAVIS_OS_NAME}  # test package creation
-# cd ../spinalcordtoolbox_v*
-# yes | ./install_sct  # test installation of package
-
+if [[ "$BRANCH" == "master" ]]; then
+  echo "*** Running batch_processing.sh ***"
+  ./batch_processing.sh
+fi
```

Comment:
I was hoping that ./ci.sh could be more agnostic than this.

In https://github.com/neuropoly/spinalcordtoolbox/issues/2674 I mused about adopting the pattern in https://docs.travis-ci.com/user/speeding-up-the-build/#parallelizing-your-builds-across-virtual-machines of

```
env:
  - SUITE=integration_tests
  - SUITE=unit_tests
  - SUITE=lint
  - SUITE=super_integration_tests # i.e. batch_processing.sh

...

script: "./travis.sh $SUITE"
```

Looking now I'm not confident I understand how `env` interacts with `matrix: include: ...`. We probably want batch_processing.sh to run on the same cases as `if: branch = release or type = cron`, but would want `SUITE=lint` to only run on one VM (say, Ubuntu). Is there a good way to build that kind of cross-product in Travis on top of what we already have in there?

Reply:
I agree, I also investigated parallelization, though I investigated the "distribute on multiple cores" approach (use of stages) vs the "distribute on multiple vm" approach that you are proposing (see https://github.com/neuropoly/spinalcordtoolbox/issues/2888#issuecomment-689585541).

Your approach would be more environment agnostic which I think is a good idea. I propose that we continue that investigation in a separate issue (created #2923 ).